packages/core/src/Utils/Language/language.js

```diff
@@ -18,6 +20,16 @@ export const changeLanguage = key => {
     }
 
     WS.setSettings(request).then(() => {
-        window.location.replace(getURL(key));
+        const new_url = new URL(window.location.href);
+        if (key === 'EN') {
+            new_url.searchParams.delete('lang');
+        } else {
+            new_url.searchParams.set('lang', key);
+        }
+        window.history.pushState({ path: new_url.toString() }, '', new_url.toString());
+        ChangeLanguageTranslation(key, () => {
```

Comment:
let's change the function name(ChangeLanguageTranslation) as camelCase as per our naming convetion.